backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/singleselectchipgroup/BpkSingleSelectChipGroup.kt

```diff
@@ -0,0 +1,89 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.singleselectchipgroup
+
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+import net.skyscanner.backpack.compose.chip.BpkChip
+import net.skyscanner.backpack.compose.chip.BpkChipStyle
+import net.skyscanner.backpack.compose.singleselectchipgroup.SingleSelectChipGroupData.ChipGroupType
+import net.skyscanner.backpack.compose.singleselectchipgroup.SingleSelectChipGroupData.ChipItem
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun BpkSingleSelectChipGroup(
+    chips: List<ChipItem>,
+    selectedIndex: Int,
+    onItemClicked: (ChipItem) -> Unit,
+    modifier: Modifier = Modifier,
+    style: BpkChipStyle = BpkChipStyle.Default,
+    type: ChipGroupType = ChipGroupType.RAIL,
+) {
+    when (type) {
+        ChipGroupType.RAIL -> {
+            LazyRow(
+                modifier = modifier,
+                state = rememberLazyListState(),
+            ) {
+                itemsIndexed(items = chips) { index, chip ->
+                    ChipItem(chip, index == selectedIndex, style) {
+                        onItemClicked.invoke(chips[index])
+                    }
+                }
+            }
+        }
+
+        ChipGroupType.WRAP -> {
+            FlowRow(
+                modifier = modifier,
+            ) {
+                chips.forEachIndexed { index, chip ->
+                    ChipItem(chip, index == selectedIndex, style) {
+                        onItemClicked.invoke(chips[index])
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun ChipItem(chip: ChipItem, selected: Boolean, style: BpkChipStyle, onSelectedChange: () -> Unit) {
+    BpkChip(
+        modifier = Modifier
+            .padding(PaddingValues(BpkSpacing.Sm))
+            .semantics { role = Role.RadioButton },
```

Comment:
when I tested the accessibility, it doesn't mention that its a radio button, it only says that its unselected item in a list and double press to toggle it, however if I set it as a checkbox/switch it mentions the type of the chip as a checkbox/switch. do you think that's enough? otherwise we may need to add a content description to explain more.

Reply:
got it - if that's how the built in component are announcing it let's leave it at that rather than making things custom. interesting that they don't announce it, though!

Reply:
I remember adding that on Friday , but it kept saying the same sentence, no difference, I will add it in case it is treated differently when using different language than English

Reply:
I think it might be an OS specific thing - I'm testing on an API 33 pixel 4 emulator and it announces radiobutton + item x of y